.github/contributing.md

```diff
@@ -0,0 +1,245 @@
+# Livtours Contribution Code Standards
+
+Livtours follows a series of operating procedures and conventions when contributing code to the codebase. These are outlined below.
+
+## Branch creation and naming conventions utilised
+
+Upon tackling a new issue in our codebase,
+
+1. Checkout and pull `dev` branch (ie. default branch)
+
+2. Create your new branch following the convention as follows:
+
+`{ISSUE-TYPE}/{ISSUE-ID}/{ISSUE-DESCRIPTION}`
+
+where:
+
+- ISSUE-TYPE: One of the issue type label group prefixes specified in the `GH labels`
+      documentation or currently in place on Github.
+
+      FEAT = type:feature
+      BUG = type:bug
+      CHORE = type:chore
+      REF = type:refactor
+      PROJ = type:project
+
+      (eg. FEAT, BUG, CHORE, DOCS, TEST, REF, PROJ, etc..)
+
+<br>
+
+- ISSUE-ID: This the id automatically generated when the issue is created on Github.
+
+  The id must:
+   - be unique
+   - contain a precursory hashtag
+
+  This can be found here:
+
+  ![](./resources/doc-images/issue-id.png)
+
+<br>
+
+- ISSUE-DESCRIPTION: This is a description of the issue.
+
+  It must be:
+   - clear
+   - concise
+   - sensical
+   - self-explanatory
+
+  It must contain:
+   - ONLY lowercase characters
+   - "-" as substitutes for spaces in multi-word descriptions
+
+  <br>
+
+  A few examples of sample branches:
+    1. `FEAT/#567/format-paragraphs-acc-page`
+    2. `REF/#78/modify-cart-request-logic`
+    3. `DOCS/#69/add-organisation-level-code-contribution-file`
+
+<br>
+
+## Pre-committing to Github
+
+Before committing our code to Github, there are a few considerations we need to take into account
```

Comment:
No, I actually mean this is of the personal responsibility of the contributor to ensure they have taken every precaution/action they can before pushing to Github, before even involving any reviewers. 

Furthermore, I would say before you open a PR as reviewers may find your PR before you even request them and leave review comments indicating a lot of changes are needed. The pre-review should be undertaken in the code editor by the contributor. 

If unsure and want to make a PR and have it ready, make an introductory Draft PR.

Reply:
Sorry, but I don't agree with this. Github is a space for sharing code. I don't see why I shouldn't be free to share some broken code (because I want to show a bug or a weird behaviour to a colleague, because I'm going on holiday and someone else is going to continue on my work, ...).

It's ok to ask everyone to check their stuff locally. But that in independent of pushing code to GitHub. The thing everyone should be careful about is to require reviews when everything was checked.

Moreover, some checks are better done only in CI, so it could happen that something fails in CI after a review was already asked.

Reply:
Okay, I didn't think of it in this context. Good valid point, thanks for bringing it to my attention. 

I guess your second paragraph is ideally what should be inserted into the doc. 

`Local Precursory Review` instead of `Pre-Comitting to Github`

Reply:
I think the important thing here is properly using the "draft pr" feature